### PR TITLE
Add options.escape

### PR DESCRIPTION
**Pull request by wolfgang42**
I needed this because I was word-wrapping some text that would be embedded in an SVG document. If I escaped it _before_ embedding (a) the lengths would have been wrong, and (b) an entity could be broken across two lines.

verb decided it wanted to modify some other stuff in the readme, so I let it.

### CODE REVIEW

**Comment by jonschlinkert** [comment]
thanks! reviewing now

> verb decided it wanted to modify some other stuff in the readme, so I let it.

ha, well if verb makes bad choices, please let me know lol. I'd love feedback

File: index.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ module.exports = function(str, options) {
 
   var newline = options.newline || '\n' + indent;
 
+  var escape = options.escape || function(str){return str;};
```
Comment (jonschlinkert) [inline index.js]: would you mind changing this to:

```js
function identity(str) { 
  return str; 
};
var escape = typeof options.escape === 'function'
  ? options.escape
  : identity;
```
so we can explicitly ensure that `escape` will always be a function (and not `true` or something)

File: index.js [hunk 2]
```diff
@@ -28,7 +30,7 @@ module.exports = function(str, options) {
   }
 
   var lines = str.match(re) || [];
-  var res = indent + lines.join(newline);
+  var res = indent + lines.map(function(str){return escape(str)}).join(newline);
```
Comment (jonschlinkert) [inline index.js]: since `escape should always be a function, here we can just do:

```js
var res = indent + lines.map(escape).join(newline);
```

**Comment by jonschlinkert** [comment]
looks great, and it's a great idea. would you mind making the proposed changes? If not, that's okay I can merge as-is and make the changes - I'm grateful for the pr.

File: index.js (hunk 2 above)
Comment (wolfgang42) [inline index.js]: I was trying to figure out how to make that map call look better--somehow it didn't occur to me that it was already a function!

**Comment by wolfgang42** [comment]
It's not that verb made bad choices, no. I just thought I'd explain why the diff of Readme.md had some unrelated changes in it.

**Comment by wolfgang42** [comment]
@jonschlinkert Have you had a chance to look at the changes I made?

**Comment by jonschlinkert** [comment]
thanks for the reminder!

**Pull request merged**